server.js

```diff
@@ -139,8 +140,31 @@ const auspicePaths = [
   "/community/:user/:repo/*",
 ];
 
+/*
+ * Redirect to translations of narratives if the client has
+ * set language preference and the translation is available
+ *
+ * Currently only used for nCoV narratives and requires the
+ * language code to be the second to last part of the url.
+ */
+const redirectToTranslatedNarratives = async (req, res) => {
+  const acceptedLanguages = req.acceptsLanguages();
+  if (acceptedLanguages && !acceptedLanguages[0].startsWith("en")) {
```

Comment:
We will have a much better chance of successfully identifying a matching language if we support the full `Accept-Language` negotiation standard. This includes matching on preferred languages other than the first as well as matching our language-only codes (`de`, `zh`) to language-region codes (`de-CH`, `de-DE`, `zh-CN`), which are commonly found in `Accept-Language` headers sent by browsers.

Taking a look at NPM, I expect the [accept-language](https://www.npmjs.com/package/accept-language) package will make this match easy to make compliant.

Reply:
Looking at the [express-request-language](https://www.npmjs.com/package/express-request-language) which is the express middleware of accept-language, it only returns a single language instead of a list of _all_ accepted languages. So, using this we would match language-region codes, but only take into account the users first language preference.

Reply:
1. I don't think the express-request-language middleware is appropriate here since the list of supported languages isn't static.

2. A single language match is what we want, yes? Given a list of all a user's preferred languages, in order of preference, and a list of the languages we support (e.g. have narratives written in), we want to find the supported language which is most preferred by the user.

Reply:
I thought we wanted to serve the user's second choice in language if the first language translation was not available?

Reply:
(@joverlee521 and I just talked about this in person, but for posterity and anyone reading, I'll finish the reply I was in the middle of writing.)

Yes, exactly. :-)

When I say "we want to find the supported language which is most preferred by the user", I mean that if we support (in no particular order) A, B, C and the user prefers (in order) D, E, C, A then the ranked intersection of those is C, A and we want to use C. Using accept-language to do this would look like:

```js
acceptLanguages.languages(['A', 'B', 'C']);
acceptLanguages.get('D, E, C, A') → C
```

If the intersection is the empty set, then `acceptLanguages.get` returns nothing and we pick a default (probably English).